### DISCUSSION TITLE
How can I implement a "trending now" behavior?

### DISCUSSION THREAD

**Discussion by titong0** [vercel/next.js#46212] [Help] [answered]
Essentially, my site uses static generation with fallback in order to do something like SSR without actually doing heavy network operations everytime a user requests the page.

What I am trying to achieve is storing the amount of times certain parameters are passed to a route. (either with Middleware or sending a request to '.../api/track' on load).

My first idea was to create an empty array and, on every request, push the parameter to it. Something like

VISITED_PAGES.push({id: someId, count: 1})

And on later requests check if the id is already present in the array, in which case it would just increment the count.

This method did not work as the VISITED_PAGES array seemed to be defined again and again on every request, which meant it was always empty. There is no need for the count to be kept everytime I deploy a change to the site.

**Comment by icyJoseph** [comment, marked as answer]
Right, you got bitten by serverless deployments. 

Take a look at this video, https://youtu.be/jv4Soz08gBw?t=1270, each request may or may not hit the same API route instance, those who belong to the same instance have the same border color.

So that means that upon receiving a request a new instance of your code spins up, does its work, and after a timeout it dies out, all of its JavaScript context goes away with it. 

It is not guaranteed that two successive requests reach to an already existing instance, it is really up to the load balancing and whatever else sits in front of your app (Vercel/Netlify/AWS Amplify, etc. infra), that's just how serverless works.

So you gotta use a DB service, and post to it, like supabase for example, which has a nice free tier. Though the DB shuts down if its unused for some time.

**Reply by titong0** [reply]
So there is no way of keeping any variable in memory? I don't want to keep it forever like in a db it's just a temporary variable, when the site shuts down it doesn't need to be kept.

**Reply by icyJoseph** [reply]
I mean, if the function handling the `api/track`, lives long enough... but in free tier, those live max 10 seconds.

**Reply by titong0** [reply]
Damn that sucks, Thanks for the help!

**Reply by icyJoseph** [reply]
Try one of the Vercel integrations, they have nice options there to be able to have short lived data.

You could maybe use the https://countapi.xyz/ too ~

**Reply by titong0** [reply]
Thanks! I'll look into supabase, seems pretty good for my usecase.
